dffml/util/data.py

```diff
@@ -123,3 +130,47 @@ def export_dict(**kwargs):
         elif isinstance(kwargs[key], list):
             kwargs[key] = export_list(kwargs[key])
     return kwargs
+
+def explore_directories(path_dict:str):
+    """
+    path_dict :
+        {
+            "hello":"there",
+            "deadbeef" : path_to_deadbeef
+        }
+    """
+    for key,val in path_dict.items():
+        if(os.path.isdir(val)):
+            temp_path_dict = {}
+            for root,dirs,files in os.walk(val):
```

Comment:
If you use `pathlib.Path().rglob()` on the top level directory it will list all files and directories all the way down in one loop.